sigoptlite/sigoptlitetest/endpoints/test_observation.py

```diff
@@ -9,70 +12,493 @@
+      assert observation.values[0].value == i
+
+    observations = connection.experiments(e.id).observations().fetch()
+    assert observations.count == num_observations
+    assert len(observations.data) == num_observations
+    for i, obs in enumerate(observations.data):
+      assert obs.assignments["d1"] == 0.5 + i
+      assert obs.assignments["i1"] == 10 + i
+      assert obs.values[0].value == i
+
+  @pytest.mark.parametrize(
+    "feature",
+    [
+      "constraints",
+      "conditionals",
+      "default",
+      "multisolution",
+      "priors",
+    ],
+  )
+  def test_single_metric_observations(self, connection, feature):
+    num_observations = 4
+    experiment_meta = self.get_experiment_feature(feature)
+
+    e = connection.experiments().create(**experiment_meta)
+    assert len(e.metrics) == 1
+    metric_name = e.metrics[0].name
+
+    assignments_list = []
+    values_list = []
+    for i in range(num_observations):
+      suggestion = connection.experiments(e.id).suggestions().create()
+      assignments = dict(suggestion.assignments)
+      values = [{"name": metric_name, "value": i}]
+      observation = self.create_observation_endpoint(
+        connection=connection,
+        experiment=e,
+        assignments=assignments,
+        values=values,
+        suggestion_id=suggestion.id,
+        use_suggestion_id=i % 2,
+      )
+      for p in assignments.keys():
+        assert observation.assignments[p] == assignments[p]
+      assert len(observation.values) == 1
+      assert observation.values[0].name == metric_name
+      assert observation.values[0].value == i
+      assignments_list.append(assignments)
+      values_list.append(values)
+
+    observations = connection.experiments(e.id).observations().fetch()
+    assert observations.count == num_observations
+    assert len(observations.data) == num_observations
+    for obs, params, vals in zip(observations.data, assignments_list, values_list):
+      for p in params.keys():
+        assert obs.assignments[p] == params[p]
+      assert obs.values[0].name == vals[0]["name"]
+      assert obs.values[0].value == vals[0]["value"]
+
+  def test_assignments_observations_log_transform_bounds(self, connection):
+    experiment_meta = self.get_experiment_feature("default")
+    experiment_meta["parameters"] = [
+      dict(name="x1", type="double", bounds=dict(min=1, max=100), transformation="log"),
+    ]
+    e = connection.experiments().create(**experiment_meta)
+
+    with pytest.raises(SigOptException):
+      connection.experiments(e.id).observations().create(
+        assignments=dict(x1=-1),
+        values=[{"name": "y1", "value": 1.23}],
+      )
+
+    o = (
+      connection.experiments(e.id)
+      .observations()
+      .create(
+        assignments=dict(x1=0.5),
+        values=[{"name": "y1", "value": 1.23}],
+      )
+    )
+    assert o is not None
+
+  def test_out_of_bounds_observations_int_and_doubles(self, connection):
```

Comment:
I think this was mentioned earlier by @harvey-intel , but we do allow out of bounds observations for ints and doubles I think, so perhaps this test can be removed and the error handling for this can be removed as well?

Reply:
there's no error here, it's a statement that this is on purpose.